diff --git a/cmd/linstor-csi/linstor-csi.go b/cmd/linstor-csi/linstor-csi.go
--- a/cmd/linstor-csi/linstor-csi.go	(revision 863eaeeb6bfd4402263991b9bed0a265c5061d62)
+++ b/cmd/linstor-csi/linstor-csi.go	(revision e55e9bfc54913374323c358175b63781e6e51537)
@@ -42,7 +42,7 @@
 	var (
 		lsEndpoint            = flag.String("linstor-endpoint", "", "Controller API endpoint for LINSTOR")
 		lsSkipTLSVerification = flag.Bool("linstor-skip-tls-verification", false, "If true, do not verify tls")
-		csiEndpoint           = flag.String("csi-endpoint", "unix:///var/lib/kubelet/plugins/linstor.csi.linbit.com/csi.sock", "CSI endpoint")
+		csiEndpoint           = flag.String("csi-endpoint", "unix:///var/lib/kubelet/plugins/replicated.csi.storage.deckhouse.io/csi.sock", "CSI endpoint")
 		node                  = flag.String("node", "", "Node ID to pass to node service")
 		logLevel              = flag.String("log-level", "info", "Enable debug log output. Choose from: panic, fatal, error, warn, info, debug")
 		rps                   = flag.Float64("linstor-api-requests-per-second", 0, "Maximum allowed number of LINSTOR API requests per second. Default: Unlimited")
diff --git a/pkg/linstor/const.go b/pkg/linstor/const.go
--- a/pkg/linstor/const.go	(revision 863eaeeb6bfd4402263991b9bed0a265c5061d62)
+++ b/pkg/linstor/const.go	(revision e55e9bfc54913374323c358175b63781e6e51537)
@@ -24,7 +24,7 @@

 const (
 	// DriverName is the name used in CSI calls for this driver.
-	DriverName = "linstor.csi.linbit.com"
+	DriverName = "replicated.csi.storage.deckhouse.io"

 	// LinstorBackupKVName is the name of the KV store used to map L2L backups to local snapshot names
 	LinstorBackupKVName = "csi-backup-mapping"
