Subject: [PATCH] linbit labels changes
---
Index: pkg/topology/topology.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pkg/topology/topology.go b/pkg/topology/topology.go
--- a/pkg/topology/topology.go	(revision a0322d71ce98d92bdd4c357c9a366e149985dc5f)
+++ b/pkg/topology/topology.go	(revision 4e4be7921b9e42f5f2ed155fc99014ac565e7f21)
@@ -47,11 +47,11 @@
 const (
 	// LinstorNodeKey refers to a node running the LINSTOR csi node service
 	// and the linstor Satellite and is therefore capable of hosting LINSTOR volumes.
-	LinstorNodeKey = "linbit.com/hostname"
+	LinstorNodeKey = "storage.deckhouse.io/sds-replicated-volume-hostname"

 	// LinstorStoragePoolKeyPrefix is the prefix used when specifying the available storage
 	// pools on a node via CSI topology keys.
-	LinstorStoragePoolKeyPrefix = "linbit.com/sp-"
+	LinstorStoragePoolKeyPrefix = "storage.deckhouse.io/sds-replicated-volume-sp-"

 	// LinstorStoragePoolValue is the value assigned to the storage pool label, given that the node has access to the
 	// storage pool.
