From 0aa2aae92a1c7d5e2f92e4e865309ef1928d6a99 Mon Sep 17 00:00:00 2001
From: "v.oleynikov" <vasily.oleynikov@flant.com>
Date: Wed, 6 Nov 2024 12:57:06 +0300
Subject: [PATCH] Removed commit checks

Signed-off-by: v.oleynikov <vasily.oleynikov@flant.com>
---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 0646cedd7..05ae776c6 100644
--- a/Makefile
+++ b/Makefile
@@ -82,14 +82,14 @@ debrelease:
 check-all-committed:
 ifneq ($(FORCE),1)
 	if ! tmp=$$(git diff --name-status HEAD 2>&1) || test -n "$$tmp" ; then \
-		echo >&2 "$$tmp"; echo >&2 "Uncommitted changes"; exit 1; \
+		echo >&2 "$$tmp"; echo >&2 "Uncommitted changes"; \
 	fi
 	if ! grep -q "^linstor-server ($(DEBVERSION)" debian/changelog ; then \
-		echo >&2 "debian/changelog needs update"; exit 1; \
+		echo >&2 "debian/changelog needs update"; \
 	fi
 	for df in "$(DOCKERFILE_CONTROLLER)" "$(DOCKERFILE_SATELLITE)"; do \
 		if test "$$(grep -c "ENV LINSTOR_VERSION $(VERSION)" "$$df")" -ne 2 ; then \
-			echo >&2 "LINSTOR_VERSION in $$df needs update"; exit 1; \
+			echo >&2 "LINSTOR_VERSION in $$df needs update"; \
 		fi; \
 	done
 endif
--
2.43.0
