diff --git a/cmd/linstor-scheduler-extender/linstor-scheduler-extender.go b/cmd/linstor-scheduler-extender/linstor-scheduler-extender.go
index 9f61882..337cedb 100644
--- a/cmd/linstor-scheduler-extender/linstor-scheduler-extender.go
+++ b/cmd/linstor-scheduler-extender/linstor-scheduler-extender.go
@@ -24,6 +24,11 @@ import (
 	_ "github.com/piraeusdatastore/linstor-scheduler-extender/pkg/driver"
 )

+const (
+	defaultCertFile string = "/etc/sds-replicated-volume-scheduler-extender/certs/tls.crt"
+	defaultKeyFile  string = "/etc/sds-replicated-volume-scheduler-extender/certs/tls.key"
+)
+
 var ext *extender.Extender

 func main() {
@@ -62,7 +67,7 @@ func run(c *cli.Context) {

 	verbose := c.Bool("verbose")
 	if verbose {
-		log.SetLevel(log.DebugLevel)
+		log.SetLevel(log.TraceLevel)
 	}

 	config, err := rest.InClusterConfig()
@@ -91,6 +96,9 @@ func run(c *cli.Context) {
 	ext = &extender.Extender{
 		Driver:   d,
 		Recorder: recorder,
+		CertFile: defaultCertFile,
+		KeyFile:  defaultKeyFile,
+		LogLevel: log.GetLevel(),
 	}

 	if err = ext.Start(); err != nil {
