Subject: [PATCH] add checksum
---
Index: main.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/main.go b/main.go
--- a/main.go	(revision 7bef2e7976a455550bce2533487c635f20390ccf)
+++ b/main.go	(revision f5cdcdd1e45367eba43b5acf21b34de1ebc7859b)
@@ -2,8 +2,10 @@

 import (
 	"bytes"
+	"crypto/md5"
 	"crypto/sha1"
 	"encoding/base64"
+	"encoding/hex"
 	"errors"
 	"flag"
 	"fmt"
@@ -60,6 +62,11 @@
 	logger *zap.Logger
 }

+func GetMD5Hash(text []byte) string {
+	hash := md5.Sum(text)
+	return hex.EncodeToString(hash[:])
+}
+
 func main() {
 	flag.Parse()

@@ -280,8 +287,7 @@
 		return nil, fmt.Errorf("Could not extract tarball: %v", err)
 	}

-	// '_' is never an existing md5sum
-	cocciPath := filepath.Join(dir, intarballName, "drbd", "drbd-kernel-compat", "cocci_cache", "_")
+	cocciPath := filepath.Join(dir, intarballName, "drbd", "drbd-kernel-compat", "cocci_cache", GetMD5Hash(compath))
 	if err := os.MkdirAll(cocciPath, 0755); err != nil {
 		return nil, fmt.Errorf("Could not create cocci dir: %v", err)
 	}