diff --git a/drbd-reactor.spec b/drbd-reactor.spec
index d3bd88c..fc2e1c4 100644
--- a/drbd-reactor.spec
+++ b/drbd-reactor.spec
@@ -11,7 +11,6 @@ License:	ASL 2.0
 URL:		https://www.github.com/LINBIT/drbd-reactor
 Source0:	https://pkg.linbit.com/downloads/drbd/utils/%{name}-%{tarball_version}.tar.gz

-BuildRequires:	systemd
 # While most pkgs I looked at have bash-completion as BuildRequires, I don't think we need it
 # looks like it would only help for cmake or pkgconfig projects.
 # BuildRequires:	bash-completion
